clients/src/main/java/org/apache/kafka/common/requests/JoinGroupResponse.java

```diff
@@ -65,7 +66,17 @@ public JoinGroupResponse(short errorCode,
                              String memberId,
                              String leaderId,
                              Map<String, ByteBuffer> groupMembers) {
-        super(new Struct(CURRENT_SCHEMA));
+        this(CURRENT_VERSION, errorCode, generationId, groupProtocol, memberId, leaderId, groupMembers);
```

Comment:
Since the version is supposed to be tied to the request version, having this constructor seems risky. Are you just using this approach since both versions of the response schemas are the same so it's not an issue at the moment?